File: app/src/main/java/ru/ulstu_team/ulstuschedule/ui/cathedries/CathedriesActivity.java
package ru.ulstu_team.ulstuschedule.ui.cathedries;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.LinearLayoutManager;
import android.view.View;

import javax.inject.Inject;

import ru.ulstu_team.ulstuschedule.R;
import ru.ulstu_team.ulstuschedule.data.model.Cathedra;
import ru.ulstu_team.ulstuschedule.databinding.ActivityCathedriesBinding;
import ru.ulstu_team.ulstuschedule.ui.base.BaseActivity;

public class CathedriesActivity extends BaseActivity implements CathedriesMvpView {

    @Inject CathedriesPresenter mPresenter;
    @Inject CathedriesAdapter mAdapter;
    private ActivityCathedriesBinding b;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        b = DataBindingUtil.setContentView(this, R.layout.activity_cathedries);
        getActivityComponent().inject(this);

        b.rvCathedries.setLayoutManager(new LinearLayoutManager(this));
        b.rvCathedries.setHasFixedSize(true);

        mPresenter.attachView(this);
        mPresenter.loadCathedries();

        b.srlRefresh.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                mPresenter.loadCathedries();
                b.srlRefresh.setRefreshing(false);
            }
        });
    }

    @Override
    public void showCathedries(Cathedra[] cathedries) {
        mAdapter.setCathedries(cathedries);
        b.rvCathedries.setAdapter(mAdapter);
    }

    @Override
    public void showEmptyList() {

    }

    @Override
    public void showError() {
        Snackbar.make(b.drawerLayout, "Возниикла ошибка", Snackbar.LENGTH_LONG)
                .setAction("Повторить", new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        mPresenter.loadCathedries();
                    }
                }).show();
    }
}

File: app/src/main/java/ru/ulstu_team/ulstuschedule/ui/cathedries/CathedriesActivity.kt
package ru.ulstu_team.ulstuschedule.ui.cathedries

import android.os.Bundle
import android.support.design.widget.Snackbar
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_layout.*
import kotlinx.android.synthetic.main.cathedries_content.*
import kotlinx.android.synthetic.main.toolbar.*
import ru.ulstu_team.ulstuschedule.R
import ru.ulstu_team.ulstuschedule.data.model.Cathedra
import ru.ulstu_team.ulstuschedule.ui.base.BaseActivity
import java.util.*
import javax.inject.Inject

class CathedriesActivity : BaseActivity(), CathedriesMvpView {

    @Inject internal lateinit var mPresenter: CathedriesPresenter
    @Inject internal lateinit var mAdapter: CathedriesAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityComponent.inject(this)
        setContentView(R.layout.activity_layout)
        contentLayout = R.layout.cathedries_content
        toolbarLayout = R.layout.toolbar
        toolbar.title = getString(R.string.cathedries)
        setSupportActionBar(toolbar)

        rvCathedries.layoutManager = LinearLayoutManager(this)
        rvCathedries.itemAnimator = DefaultItemAnimator()
        rvCathedries.setHasFixedSize(true)
    }

    override fun onStart() {
        super.onStart()
        mPresenter.attachView(this)
        loadCathedries()

        srlRefresh.setOnRefreshListener { loadCathedries() }
    }

    private fun loadCathedries() {
        val facultyId = intent.getIntExtra("FacultyId", -1)
        if (facultyId <= 0)
            mPresenter.loadCathedries()
        else
            mPresenter.loadCathedries(facultyId)
    }

    override fun showCathedries(cathedries: List<Cathedra>) {
        mAdapter.setCathedries(cathedries)
        rvCathedries.adapter = mAdapter
        srlRefresh.isRefreshing = false
    }

    override fun showEmptyList() {
        mAdapter.setCathedries(ArrayList<Cathedra>())
        srlRefresh.isRefreshing = false
    }

    override fun showError() {
        Snackbar.make(drawer_layout, "Возниикла ошибка", Snackbar.LENGTH_LONG)
                .setAction("Повторить") { loadCathedries() }
                .show()
        srlRefresh.isRefreshing = false
    }
}
